src/libraries/System.Security.Cryptography.Cose/src/System/Security/Cryptography/Cose/CoseHeaderMap.cs

```diff
@@ -4,19 +4,28 @@
 using System.Collections;
 using System.Collections.Generic;
 using System.Diagnostics;
-using System.Diagnostics.CodeAnalysis;
 using System.Formats.Cbor;
 
 namespace System.Security.Cryptography.Cose
 {
+    /// <summary>
+    /// Represents a bucket of header parameters of a COSE message.
+    /// </summary>
     public sealed class CoseHeaderMap : IDictionary<CoseHeaderLabel, CoseHeaderValue>, IReadOnlyDictionary<CoseHeaderLabel, CoseHeaderValue>
     {
         private static readonly CoseHeaderMap s_emptyMap = new CoseHeaderMap(isReadOnly: true);
 
         private readonly Dictionary<CoseHeaderLabel, CoseHeaderValue> _headerParameters = new Dictionary<CoseHeaderLabel, CoseHeaderValue>();
 
+        /// <summary>
+        /// Gets a value that indicates whether the map cannot be modified.
```

Comment:
IList.IsReadOnly just uses read-only in the docs (probably to avoid the mid-summary negative).

```suggestion
        /// Gets a value that indicates whether the header map is read-only.
```